### ISSUE TITLE
Separate Helm charts for CRDs

### ISSUE THREAD

**Issue by lsoica** [argoproj/argo-helm#1893]
### Is your feature request related to a problem?

In case of multi tenant k8s cluster deployments, the CRDs can only be applied by a cluster admin and the rest of the deployment can be handled by regular/tenant users where cluster wide access is not required.

Has the option of publishing dedicated charts for the CRDs been discussed? Or another alternative that works better in case of multi-tenant clusters?

### Related helm chart

argo-cd

### Describe the solution you'd like

Separate charts released with only the CRDs.

**Comment by yu-croco** [comment]
Hi @lsoica,

> Has the option of publishing dedicated charts for the CRDs been discussed? 

As far as I remember, I hope https://github.com/argoproj/argo-helm/issues/1785 would answer against your question. 🤔

**Comment by lsoica** [comment]
@yu-croco yes, different chart, same problem. Nothing argo specific here, just a problem that wasn't yet solved by the ecosystem afaik. What I'm curios to understand is how do people currently handle multi tenant clusters when they are not admins anymore but they do have CRDs as part of their charts. With this approach of embedding the CRDs as part of the service chart, I don't see how could one deliver just the CRDs to a cluster admin that has the rights to apply them.

**Comment by pdrastil** [comment]
@lsoica - We run shared responsibility model - infra team provides clusters + core shared infrastructure with all CRDs. Users are given namespaces that consume the CRDs. If they need new service with CRDs it's delegated to infra team to integrate it with the cluster.

**Comment by lsoica** [comment]
Makes sense. When users own/deliver services that have their own CRDs, how do you package, release, ship the CRDs?

**Comment by pdrastil** [comment]
The point is that users do not install their own CRDs so they do not cause havoc in multi-tenant environment. They use only namespace scoped resources. If they develop custom K8s operators we have special clusters just for that where they are cluster admin (very small pool of people) or they are given single tenant cluster where they take full responsibility of patching, provisioning and keeping it healthy for their own tenants.

**Comment by lsoica** [comment]
Got it, thanks. For me it's a bit of a different setup where I don't own the actual K8S cluster, I'm just a tenant, a regular user, and I have services that need CRDs. ArgoCD included. So I have to either use the ArgoCD that's already there, or bring my own, in which case, I have to hand over the CRDs to the cluster admin. So I have to build some sort of versioned package.

**Comment by pdrastil** [comment]
If Argo CD is there and you don't want to use provided one for whatever reason you can install your own without CRDs in namespaced mode. You will be limited to version of Argo CD that is maintained by your cluster admin as they are upgrading CRDs for the shared instance.
